aws-predictions/src/test/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformersTest.java

```diff
@@ -0,0 +1,259 @@
+                .withBlockType(BlockType.TABLE)
+                .withConfidence(random.nextFloat())
+                .withGeometry(randomGeometry())
+                .withRelationships(new Relationship()
+                        .withIds(cellBlock.getId()));
+
+        // Construct a map to act as a graph
+        Map<String, Block> blockMap = new HashMap<>();
+        blockMap.put(cellTextBlock.getId(), cellTextBlock);
+        blockMap.put(cellSelectionBlock.getId(), cellSelectionBlock);
+        blockMap.put(cellBlock.getId(), cellBlock);
+        blockMap.put(tableBlock.getId(), tableBlock);
+
+        // Test table block conversion
+        Table table = TextractResultTransformers.fetchTable(tableBlock, blockMap);
+        assertEquals(1, table.getCells().size());
+        assertEquals(tableBlock.getConfidence(), table.getConfidence(), DELTA);
+        assertEquals(1, table.getRowSize());
+        assertEquals(1, table.getColumnSize());
+
+        // Test cell block conversion
+        Cell cell = table.getCells().iterator().next();
+        assertEquals(cellTextBlock.getText(), cell.getText());
+        assertEquals(cellBlock.getConfidence(), cell.getConfidence(), DELTA);
+        assertTrue(cell.isSelected());
+        assertEquals(cellBlock.getRowIndex() - 1, cell.getRow());
+        assertEquals(cellBlock.getColumnIndex() - 1, cell.getColumn());
+    }
+
+    /**
+     * Tests that the graph of related blocks from Textract is properly
+     * converted to an Amplify key-value pair item.
+     */
+    @Test
+    public void testKeyValueConversion() {
+        Block valueTextBlock = new Block()
+                .withId(RandomString.string())
+                .withText(RandomString.string());
+        Block keyTextBlock = new Block()
+                .withId(RandomString.string())
+                .withText(RandomString.string());
+        Block valueBlock = new Block()
+                .withId(RandomString.string())
+                .withBlockType(BlockType.KEY_VALUE_SET)
+                .withEntityTypes(EntityType.VALUE.toString())
+                .withRelationships(new Relationship()
+                        .withIds(valueTextBlock.getId()));
+        Block keyBlock = new Block()
+                .withId(RandomString.string())
+                .withBlockType(BlockType.KEY_VALUE_SET)
+                .withEntityTypes(EntityType.KEY.toString())
+                .withConfidence(random.nextFloat())
+                .withGeometry(randomGeometry())
+                .withRelationships(new Relationship()
+                        .withIds(keyTextBlock.getId(), valueBlock.getId()));
+
+        // Construct a map to act as a graph
+        Map<String, Block> blockMap = new HashMap<>();
+        blockMap.put(valueTextBlock.getId(), valueTextBlock);
+        blockMap.put(keyTextBlock.getId(), keyTextBlock);
+        blockMap.put(valueBlock.getId(), valueBlock);
+        blockMap.put(keyBlock.getId(), keyBlock);
```

Comment:
Whenever you duplicate an operation 2-3 times, you should move to a control structure.
```
List<Block> blocks = Arrays.asList(valueTextBlock, keyTextBlock, valueBlock, keyBlock);
Map<String, Block> blocksById = new HashMap<>();
for (Block block : blocks) {
    blocksById.put(block.getId(), block);
}
```
Or:
```
Map<String, Block> blocksById = new HashMap<>();
Observable.fromArray(valueTextBlock, keyTextBlock, valueBlock, keyBlock)
    .blockingForEach(block -> blocksById.put(block.getId(), block));
```

Reply:
the second solution seems nice! I'll do that